### PR TITLE
Enforce `use_key_in_widget_constructors` and `file_names` lints

### PR DESCRIPTION
**Pull request by domesticmouse**
fixes: https://github.com/flutter/samples/issues/807

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: add_to_app/books/flutter_module_books/lib/api.dart [hunk 1]
```diff
@@ -94,6 +94,7 @@ abstract class FlutterBookApi {
       if (api == null) {
         channel.setMessageHandler(null);
       } else {
+        // ignore: avoid_types_on_closure_parameters
         channel.setMessageHandler((Object? message) async {
```
Comment (johnpryan) [inline add_to_app/books/flutter_module_books/lib/api.dart:98]: Is the type necessary here?

File: platform_design/analysis_options.yaml [hunk 2]
```diff
@@ -17,4 +17,5 @@ linter:
     test_types_in_equals: true
     throw_in_finally: true
     unnecessary_statements: true
+    # Tests fail if we enforce `use_key_in_widget_constructors`
```
[the remark this answers is not in the document]
Comment (domesticmouse) [reply platform_design/analysis_options.yaml:20]: I honestly don't know. 

Probably worth creating a PR with just this project updated to conform to this lint and flag it to the original author to get his or her understanding of why the tests are breaking.

File: add_to_app/books/flutter_module_books/lib/api.dart (hunk 1 above)
Comment (domesticmouse) [reply add_to_app/books/flutter_module_books/lib/api.dart:98]: I don't recall, but I usually leave types in when things break on removal of said type declaration

File: experimental/web_dashboard/lib/src/widgets/dialogs.dart [hunk 3]
```diff
@@ -11,10 +11,12 @@ import '../app.dart';
...
 class NewCategoryDialog extends StatelessWidget {
+  const NewCategoryDialog({Key key}) : super(key: key);
+
   @override
   Widget build(BuildContext context) {
-    return SimpleDialog(
-      title: const Text('New Category'),
+    return const SimpleDialog(
+      title: Text('New Category'),
       children: <Widget>[
```
Comment (RedBrogdon) [inline experimental/web_dashboard/lib/src/widgets/dialogs.dart:20]: I wish we had a lint just for these unnecessary `<Widget>` specifiers that used to be required.

File: platform_design/analysis_options.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (RedBrogdon) [reply platform_design/analysis_options.yaml:20]: @domesticmouse, let's create an issue for it, at the very least, so it won't get lost in the shuffle.

File: experimental/web_dashboard/lib/src/widgets/dialogs.dart (hunk 3 above)
Comment (johnpryan) [reply experimental/web_dashboard/lib/src/widgets/dialogs.dart:20]: It doesn't look like anyone's filed an issue for this yet: https://github.com/flutter/flutter/labels/p%3A%20flutter_lints

**Pull request merged**